libguard/guard_interface.cpp

```diff
@@ -75,7 +77,7 @@ const fs::path& getGuardFilePath()
 {
     if (guardFilePath.empty())
     {
-        throw std::runtime_error(
+        throw InvalidEntry(
```

Comment:
I think you can throw `GuardFileOpenFailed` which is more appropriate instead of `InvalidEntry` for this interface API.

It will be great if you added a trace as well like an exception message which is throwing.

Reply:
Changed to GuardFileOpenFailed and added trace "Guard file not initialized"